packages/itwin/tree-widget/src/e2e-tests/auth.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/* eslint-disable no-console */
+
+import { getTestAccessToken } from "@itwin/oidc-signin-tool";
+import { test as setup } from "@playwright/test";
+
+setup("authenticate", async ({ page }) => {
+  if (!process.env.IMJS_USER_EMAIL || !process.env.IMJS_USER_PASSWORD) {
+    throw new Error(`User credentials need to be supplied through IMJS_USER_EMAIL and IMJS_USER_PASSWORD environment variables.`);
+  }
+  if (!process.env.IMJS_AUTH_CLIENT_CLIENT_ID) {
+    throw new Error(`Auth client id needs to be supplied through IMJS_AUTH_CLIENT_CLIENT_ID environment variable.`);
+  }
+  // eslint-disable-next-line @itwin/no-internal
+  const accessToken = await getTestAccessToken(
+    {
+      authority: "https://qa-ims.bentley.com",
+      scope: "imodelaccess:read imodels:read realitydata:read",
+      clientId: process.env.IMJS_AUTH_CLIENT_CLIENT_ID,
+      redirectUri: "http://localhost:3000/signin-callback",
+    },
+    {
+      email: process.env.IMJS_USER_EMAIL,
+      password: process.env.IMJS_USER_PASSWORD,
+    },
+  );
+  await page.context().addCookies([
+    {
+      name: "IMJS_USER_ACCESS_TOKEN",
+      value: accessToken ?? "",
+      domain: "localhost",
+      path: "/",
+    },
```

Comment:
Do cookies get cleaned before each e2e test run?

Reply:
No. The `auth` script puts this value into a cookie and saves the browser context to a file. Then, before running the actual tests, we initialize the browser state from the saved file - this makes the cookie available to all the tests.